icu4c/source/test/intltest/unitstest.cpp

```diff
@@ -458,9 +501,53 @@ class ExpectedOutput {
     }
 };
 
+void checkOutput(UnitsTest *unitsTest, const char *msg, ExpectedOutput expected,
+                  const MaybeStackVector<Measure> &actual, double precision) {
+    IcuTestErrorCode status(*unitsTest, "checkOutput");
+    bool success = true;
+    if (expected._compoundCount != actual.length()) {
+        success = false;
+    }
+    for (int i = 0; i < actual.length(); i++) {
+        if (i >= expected._compoundCount) {
+            break;
+        }
+
+        double diff = std::abs(expected._amounts[i] -
+                               actual[i]->getNumber().getDouble(status));
+        double diffPercent =
+            expected._amounts[i] != 0 ? diff / expected._amounts[i] : diff;
```

Comment:
Why aren't you using assertEqualsNear here?

Reply:
I have added a TODO to @hugovdm

Reply:
I think this check may predate the assertEqualsNear.

@younies if you feel like it, I suggest you can simply add a "TODO(hugovdm): use assertEqualsNear" then I'll take care of it as I refactor the function... (considering the changes in this file is just reformatting).

Reply:
And I was looking at an old tab which didn't have younies' comment above. :) Thanks for the TODO!